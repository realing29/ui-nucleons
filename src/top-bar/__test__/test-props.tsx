import React from 'react';
import CrossSVG from '@sima-land/ui-quarks/icons/24x24/Stroked/cross';
import ArrowLeftSVG from '@sima-land/ui-quarks/icons/24x24/Stroked/arrow-left';
import PersonSVG from '@sima-land/ui-quarks/icons/24x24/Stroked/person';

const longText =
  'Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita ad, provident aspernatur dolore. ';

export const sizes = ['s', 'm'];

export const shortTitles = { title: 'Modal title', subtitle: 'Modal subtitle' };

export const longTitles = { title: longText.repeat(2), subtitle: longText.repeat(2) };

export const startButtons = {
  start: {
    icon: <ArrowLeftSVG />,
  },
  startSecondary: {
    icon: <PersonSVG />,
  },
};

export const endButtons = {
  end: {
    icon: <CrossSVG />,
  },
  endSecondary: {
    icon: <PersonSVG />,
  },
};

export const allButtons = {
  ...startButtons,
  ...endButtons,
};
