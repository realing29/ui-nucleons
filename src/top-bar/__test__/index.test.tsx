import React from 'react';
import { mount } from 'enzyme';
import { TopBar } from '..';
import { mapValues } from 'lodash';
import * as TestProps from './test-props';

describe('<TopBar />', () => {
  it('should render without props', () => {
    const wrapper = mount(
      <TopBar />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render as divided', () => {
    const wrapper = mount(
      <TopBar divided />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should renders with title and subtitle', () => {
    const wrapper = mount(
      <TopBar {...TestProps.shortTitles} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "size" prop', () => {
    const wrapper = mount(
      <TopBar {...TestProps.shortTitles} />
    );

    TestProps.sizes.forEach(size => {
      wrapper.setProps({ size });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should renders with start icon', () => {
    const wrapper = mount(
      <TopBar
        buttonsProps={{ start: TestProps.allButtons.start }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should renders with end icon', () => {
    const wrapper = mount(
      <TopBar
        buttonsProps={{ end: TestProps.allButtons.end }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should renders with start icons', () => {
    const wrapper = mount(
      <TopBar
        buttonsProps={TestProps.startButtons}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should renders with end icons', () => {
    const wrapper = mount(
      <TopBar
        buttonsProps={TestProps.endButtons}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should renders with all icons', () => {
    const buttonProps = mapValues(TestProps.allButtons, p => ({ ...p, onClick: jest.fn() }));

    const wrapper = mount(
      <TopBar
        buttonsProps={buttonProps}
      />
    );

    expect(wrapper).toMatchSnapshot();

    // start
    wrapper.find('.icon-button').at(0).simulate('click');
    expect(buttonProps.start.onClick).toBeCalledTimes(1);

    // start secondary
    wrapper.find('.icon-button').at(1).simulate('click');
    expect(buttonProps.startSecondary.onClick).toBeCalledTimes(1);

    // end
    wrapper.find('.icon-button').at(2).simulate('click');
    expect(buttonProps.endSecondary.onClick).toBeCalledTimes(1);

    // end secondary
    wrapper.find('.icon-button').at(3).simulate('click');
    expect(buttonProps.end.onClick).toBeCalledTimes(1);
  });
});
