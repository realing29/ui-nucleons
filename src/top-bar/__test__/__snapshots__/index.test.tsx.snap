// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should handle "size" prop 1`] = `
<TopBar
  size="s"
  subtitle="Modal subtitle"
  title="Modal title"
>
  <div
    className="modal-header size-s"
  >
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      >
        Modal title
      </div>
      <div
        className="subtitle ellipsis"
      >
        Modal subtitle
      </div>
    </div>
  </div>
</TopBar>
`;

exports[`<TopBar /> should handle "size" prop 2`] = `
<TopBar
  size="m"
  subtitle="Modal subtitle"
  title="Modal title"
>
  <div
    className="modal-header size-m"
  >
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      >
        Modal title
      </div>
      <div
        className="subtitle ellipsis"
      >
        Modal subtitle
      </div>
    </div>
  </div>
</TopBar>
`;

exports[`<TopBar /> should handle "size" prop 3`] = `
<TopBar
  size="l"
  subtitle="Modal subtitle"
  title="Modal title"
>
  <div
    className="modal-header size-l"
  >
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      >
        Modal title
      </div>
      <div
        className="subtitle ellipsis"
      >
        Modal subtitle
      </div>
    </div>
  </div>
</TopBar>
`;

exports[`<TopBar /> should render without props 1`] = `
<TopBar>
  <div
    className="modal-header size-m"
  >
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      />
    </div>
  </div>
</TopBar>
`;

exports[`<TopBar /> should renders with all icons 1`] = `
<TopBar
  buttonsProps={
    Object {
      "end": Object {
        "icon": <CrossSVG />,
        "onClick": [MockFunction],
      },
      "endSecondary": Object {
        "icon": <PersonSVG />,
        "onClick": [MockFunction],
      },
      "start": Object {
        "icon": <ArrowLeftSVG />,
        "onClick": [MockFunction],
      },
      "startSecondary": Object {
        "icon": <PersonSVG />,
        "onClick": [MockFunction],
      },
    }
  }
>
  <div
    className="modal-header size-m"
  >
    <IconButton
      icon={<ArrowLeftSVG />}
      onClick={[MockFunction]}
    >
      <button
        className="icon-button"
        onClick={[MockFunction]}
        type="button"
      >
        <ArrowLeftSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M5.414 13l5.293 5.293a1 1 0 01-1.414 1.414l-7-7a1 1 0 010-1.414l7-7a1 1 0 011.414 1.414L5.414 11H21a1 1 0 110 2H5.414z"
            />
          </svg>
        </ArrowLeftSVG>
      </button>
    </IconButton>
    <IconButton
      icon={<PersonSVG />}
      onClick={[MockFunction]}
    >
      <button
        className="icon-button"
        onClick={[MockFunction]}
        type="button"
      >
        <PersonSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M11.88 1C8.723 1 6.356 3.887 6.975 6.98l.455 2.274a4.66 4.66 0 009.138 0l.455-2.273C17.642 3.887 15.276 1 12.12 1h-.242zM8.936 6.588A3 3 0 0111.88 3h.242a3 3 0 012.941 3.588l-.454 2.274a2.66 2.66 0 01-5.216 0l-.455-2.274z"
              fillRule="evenodd"
            />
            <path
              d="M7.214 14.977a1 1 0 00-.428-1.954c-2.223.486-3.462 1.496-4.123 2.508a4.205 4.205 0 00-.573 1.311 3.43 3.43 0 00-.09.626c-.009.51 0 1.021 0 1.532a3 3 0 003 3h14a3 3 0 003-3c0-.51.007-1.021 0-1.532l-.002-.048a3.449 3.449 0 00-.088-.578 4.203 4.203 0 00-.573-1.31c-.66-1.013-1.9-2.023-4.123-2.509a1 1 0 10-.428 1.954c1.777.389 2.538 1.129 2.877 1.648.178.272.262.518.302.685a1.44 1.44 0 01.035.203V19a1 1 0 01-1 1H5a1 1 0 01-1-1v-1.484l.002-.025c.004-.035.013-.097.033-.18.04-.168.124-.414.302-.686.34-.52 1.1-1.26 2.877-1.648z"
            />
          </svg>
        </PersonSVG>
      </button>
    </IconButton>
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      />
    </div>
    <IconButton
      icon={<PersonSVG />}
      onClick={[MockFunction]}
    >
      <button
        className="icon-button"
        onClick={[MockFunction]}
        type="button"
      >
        <PersonSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M11.88 1C8.723 1 6.356 3.887 6.975 6.98l.455 2.274a4.66 4.66 0 009.138 0l.455-2.273C17.642 3.887 15.276 1 12.12 1h-.242zM8.936 6.588A3 3 0 0111.88 3h.242a3 3 0 012.941 3.588l-.454 2.274a2.66 2.66 0 01-5.216 0l-.455-2.274z"
              fillRule="evenodd"
            />
            <path
              d="M7.214 14.977a1 1 0 00-.428-1.954c-2.223.486-3.462 1.496-4.123 2.508a4.205 4.205 0 00-.573 1.311 3.43 3.43 0 00-.09.626c-.009.51 0 1.021 0 1.532a3 3 0 003 3h14a3 3 0 003-3c0-.51.007-1.021 0-1.532l-.002-.048a3.449 3.449 0 00-.088-.578 4.203 4.203 0 00-.573-1.31c-.66-1.013-1.9-2.023-4.123-2.509a1 1 0 10-.428 1.954c1.777.389 2.538 1.129 2.877 1.648.178.272.262.518.302.685a1.44 1.44 0 01.035.203V19a1 1 0 01-1 1H5a1 1 0 01-1-1v-1.484l.002-.025c.004-.035.013-.097.033-.18.04-.168.124-.414.302-.686.34-.52 1.1-1.26 2.877-1.648z"
            />
          </svg>
        </PersonSVG>
      </button>
    </IconButton>
    <IconButton
      icon={<CrossSVG />}
      onClick={[MockFunction]}
    >
      <button
        className="icon-button"
        onClick={[MockFunction]}
        type="button"
      >
        <CrossSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M4.293 19.707a1 1 0 001.414 0L12 13.414l6.293 6.293a1 1 0 001.414-1.414L13.414 12l6.293-6.293a1 1 0 00-1.414-1.414L12 10.586 5.707 4.293a1 1 0 00-1.414 1.414L10.586 12l-6.293 6.293a1 1 0 000 1.414z"
              fillRule="evenodd"
            />
          </svg>
        </CrossSVG>
      </button>
    </IconButton>
  </div>
</TopBar>
`;

exports[`<TopBar /> should renders with end icon 1`] = `
<TopBar
  buttonsProps={
    Object {
      "end": Object {
        "icon": <CrossSVG />,
      },
    }
  }
>
  <div
    className="modal-header size-m"
  >
    <div
      className="icon-button"
    />
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      />
    </div>
    <IconButton
      icon={<CrossSVG />}
    >
      <button
        className="icon-button"
        type="button"
      >
        <CrossSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M4.293 19.707a1 1 0 001.414 0L12 13.414l6.293 6.293a1 1 0 001.414-1.414L13.414 12l6.293-6.293a1 1 0 00-1.414-1.414L12 10.586 5.707 4.293a1 1 0 00-1.414 1.414L10.586 12l-6.293 6.293a1 1 0 000 1.414z"
              fillRule="evenodd"
            />
          </svg>
        </CrossSVG>
      </button>
    </IconButton>
  </div>
</TopBar>
`;

exports[`<TopBar /> should renders with end icons 1`] = `
<TopBar
  buttonsProps={
    Object {
      "end": Object {
        "icon": <CrossSVG />,
      },
      "endSecondary": Object {
        "icon": <PersonSVG />,
      },
    }
  }
>
  <div
    className="modal-header size-m"
  >
    <div
      className="icon-button"
    />
    <div
      className="icon-button"
    />
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      />
    </div>
    <IconButton
      icon={<PersonSVG />}
    >
      <button
        className="icon-button"
        type="button"
      >
        <PersonSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M11.88 1C8.723 1 6.356 3.887 6.975 6.98l.455 2.274a4.66 4.66 0 009.138 0l.455-2.273C17.642 3.887 15.276 1 12.12 1h-.242zM8.936 6.588A3 3 0 0111.88 3h.242a3 3 0 012.941 3.588l-.454 2.274a2.66 2.66 0 01-5.216 0l-.455-2.274z"
              fillRule="evenodd"
            />
            <path
              d="M7.214 14.977a1 1 0 00-.428-1.954c-2.223.486-3.462 1.496-4.123 2.508a4.205 4.205 0 00-.573 1.311 3.43 3.43 0 00-.09.626c-.009.51 0 1.021 0 1.532a3 3 0 003 3h14a3 3 0 003-3c0-.51.007-1.021 0-1.532l-.002-.048a3.449 3.449 0 00-.088-.578 4.203 4.203 0 00-.573-1.31c-.66-1.013-1.9-2.023-4.123-2.509a1 1 0 10-.428 1.954c1.777.389 2.538 1.129 2.877 1.648.178.272.262.518.302.685a1.44 1.44 0 01.035.203V19a1 1 0 01-1 1H5a1 1 0 01-1-1v-1.484l.002-.025c.004-.035.013-.097.033-.18.04-.168.124-.414.302-.686.34-.52 1.1-1.26 2.877-1.648z"
            />
          </svg>
        </PersonSVG>
      </button>
    </IconButton>
    <IconButton
      icon={<CrossSVG />}
    >
      <button
        className="icon-button"
        type="button"
      >
        <CrossSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M4.293 19.707a1 1 0 001.414 0L12 13.414l6.293 6.293a1 1 0 001.414-1.414L13.414 12l6.293-6.293a1 1 0 00-1.414-1.414L12 10.586 5.707 4.293a1 1 0 00-1.414 1.414L10.586 12l-6.293 6.293a1 1 0 000 1.414z"
              fillRule="evenodd"
            />
          </svg>
        </CrossSVG>
      </button>
    </IconButton>
  </div>
</TopBar>
`;

exports[`<TopBar /> should renders with start icon 1`] = `
<TopBar
  buttonsProps={
    Object {
      "start": Object {
        "icon": <ArrowLeftSVG />,
      },
    }
  }
>
  <div
    className="modal-header size-m"
  >
    <IconButton
      icon={<ArrowLeftSVG />}
    >
      <button
        className="icon-button"
        type="button"
      >
        <ArrowLeftSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M5.414 13l5.293 5.293a1 1 0 01-1.414 1.414l-7-7a1 1 0 010-1.414l7-7a1 1 0 011.414 1.414L5.414 11H21a1 1 0 110 2H5.414z"
            />
          </svg>
        </ArrowLeftSVG>
      </button>
    </IconButton>
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      />
    </div>
    <div
      className="icon-button"
    />
  </div>
</TopBar>
`;

exports[`<TopBar /> should renders with start icons 1`] = `
<TopBar
  buttonsProps={
    Object {
      "start": Object {
        "icon": <ArrowLeftSVG />,
      },
      "startSecondary": Object {
        "icon": <PersonSVG />,
      },
    }
  }
>
  <div
    className="modal-header size-m"
  >
    <IconButton
      icon={<ArrowLeftSVG />}
    >
      <button
        className="icon-button"
        type="button"
      >
        <ArrowLeftSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M5.414 13l5.293 5.293a1 1 0 01-1.414 1.414l-7-7a1 1 0 010-1.414l7-7a1 1 0 011.414 1.414L5.414 11H21a1 1 0 110 2H5.414z"
            />
          </svg>
        </ArrowLeftSVG>
      </button>
    </IconButton>
    <IconButton
      icon={<PersonSVG />}
    >
      <button
        className="icon-button"
        type="button"
      >
        <PersonSVG>
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              clipRule="evenodd"
              d="M11.88 1C8.723 1 6.356 3.887 6.975 6.98l.455 2.274a4.66 4.66 0 009.138 0l.455-2.273C17.642 3.887 15.276 1 12.12 1h-.242zM8.936 6.588A3 3 0 0111.88 3h.242a3 3 0 012.941 3.588l-.454 2.274a2.66 2.66 0 01-5.216 0l-.455-2.274z"
              fillRule="evenodd"
            />
            <path
              d="M7.214 14.977a1 1 0 00-.428-1.954c-2.223.486-3.462 1.496-4.123 2.508a4.205 4.205 0 00-.573 1.311 3.43 3.43 0 00-.09.626c-.009.51 0 1.021 0 1.532a3 3 0 003 3h14a3 3 0 003-3c0-.51.007-1.021 0-1.532l-.002-.048a3.449 3.449 0 00-.088-.578 4.203 4.203 0 00-.573-1.31c-.66-1.013-1.9-2.023-4.123-2.509a1 1 0 10-.428 1.954c1.777.389 2.538 1.129 2.877 1.648.178.272.262.518.302.685a1.44 1.44 0 01.035.203V19a1 1 0 01-1 1H5a1 1 0 01-1-1v-1.484l.002-.025c.004-.035.013-.097.033-.18.04-.168.124-.414.302-.686.34-.52 1.1-1.26 2.877-1.648z"
            />
          </svg>
        </PersonSVG>
      </button>
    </IconButton>
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      />
    </div>
    <div
      className="icon-button"
    />
    <div
      className="icon-button"
    />
  </div>
</TopBar>
`;

exports[`<TopBar /> should renders with title and subtitle 1`] = `
<TopBar
  subtitle="Modal subtitle"
  title="Modal title"
>
  <div
    className="modal-header size-m"
  >
    <div
      className="main-section"
    >
      <div
        className="title ellipsis"
      >
        Modal title
      </div>
      <div
        className="subtitle ellipsis"
      >
        Modal subtitle
      </div>
    </div>
  </div>
</TopBar>
`;
